force-app/main/default/aura/CandidateGrade/CandidateGrade.cmp

```diff
@@ -4,6 +4,11 @@
     <aura:attribute name="displayGrade" type="Boolean" default="false" />
     <aura:handler name="init" value="{!this}" action="{!c.doInit}"/>
 
+    <ltng:require
```

Comment:
### Error Handling: Missing transaction control in Apex
#### Severity
Error

#### Finding
This code should use transaction control to rollback partial database changes upon error. Please ensure that a savepoint is created before any database manipulation, and a rollback operation is performed every time an exception is caught.

#### Why is this a problem?
When attempting to save multiple records as part of the same operation, one should always assure that the database isn't left in an inconsistent state in case of error. Using transaction control (savepoints and rollback) allows rolling back partial changes in case of error and preserves data integrity.

#### Help and documentation

* [Salesforce Savepoints (Community Resources)](https://th3silverlining.com/2009/06/15/salesforce-savepoints/) 
* [Transaction Control (Developer Documentation)](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/langCon\_apex\_transaction\_control.htm) 
* [Improper Handling of Exceptional Conditions (CWE)](https://cwe.mitre.org/data/definitions/755.html)